core/src/main/java/discord4j/core/object/entity/BaseGuildChannel.java

```diff
@@ -60,6 +61,24 @@ public final Snowflake getGuildId() {
                         new PermissionOverwrite(getServiceMediator(), bean, getGuildId().asLong(), getId().asLong()))
                 .collect(Collectors.toSet());
     }
+    
+    @Override
+    public Mono<PermissionSet> getPermissions(Member member) {
```

Comment:
Im not sure that this calculation is correct. First, it does not take into account the ADMINISTRATOR permission. However, I think more importantly, it does not take into account the importance of ordering in the application of overwrites. Refer to the [permission overwrites section](https://discordapp.com/developers/docs/topics/permissions#permission-overwrites) in the Discord API docs.